crates/worker/src/partition_processor_manager.rs

```diff
@@ -780,6 +787,26 @@ impl PartitionProcessorManager {
         partition_table: &PartitionTable,
     ) {
         let partition_id = control_processor.partition_id;
+
+        if self
+            .max_observed_partition_configuration
+            .get(&partition_id)
+            .is_some_and(|version| version > &control_processor.partition_configuration_version)
+        {
```

Comment:
I'm a bit confused by the max observed version here. What we send from admin is the max(current, next) right? In the original design, we needed the two values to be sent separately (same in gossip).

Do you plan on leveraging PartitionState from FD in the future?

Reply:
I think you are right that this doesn't set things up 100% in the right direction because I don't distinguish between the current and next version here. I'll change it.

In the future the partition configuration information should be disseminated via gossip, yes. This could make some of the control processor messages also obsolete (only the leader message would be needed until we move the leader election to the ppm/pps).

Reply:
Let me actually remove this bit and properly address it once I have wired up the `PartitionReplicaSet` information. Otherwise, I fear that I need to introduce some special case handling (when stopping partition processors) which shouldn't be necessary once we rely on the `PartitionReplicaSet` for starting/stopping of partition processors.